src/main/cpp/chassis/SwerveChassis.cpp

```diff
@@ -292,20 +294,38 @@ void SwerveChassis::UpdateOdometry()
         auto useVision = (m_pigeon != nullptr && std::abs(GetRotationRateDegreesPerSecond()) < 720.0);
         if (useVision)
         {
-            std::optional<VisionPose> megaTag2Pose = m_vision->GetRobotPositionMegaTag2(GetYaw(), // mtAngle.Degrees(),
-                                                                                        units::angular_velocity::degrees_per_second_t(0.0),
-                                                                                        units::angle::degree_t(0.0),
-                                                                                        units::angular_velocity::degrees_per_second_t(0.0),
-                                                                                        units::angle::degree_t(0.0),
-                                                                                        units::angular_velocity::degrees_per_second_t(0.0));
-
-            if (megaTag2Pose)
+            auto hasValidRotation = true;
+            auto rotation = GetYaw();
+            if (m_pigeon->GetStickyFault_Undervoltage().GetValue()) // Brownout
             {
-                m_poseEstimator.SetVisionMeasurementStdDevs(megaTag2Pose->visionMeasurementStdDevs); // wpi::array<double, 3>(.7, .7, 9999999));
+                hasValidRotation = false;
+                m_pigeon->ClearStickyFault_Undervoltage();
```

Comment:
I don't feel like we should just clear this immediately, I think we need a hysteresis. Also I don't know if this is going to fix the problem we have. The issue is the brownout causes the pigeon to "Drift" essentially stop tracking where we are and be offset. I think if we have a brown out we need to switch to using megatag 1 instead of megatag 2. This will be more robust. This way we fix the Pigeon "Drift" problem as well as the pose problem. Maybe after we have updated with MegaTag 1, we can then clear the fault and go back to MegaTag2.

Reply:
I am using mega tag1 rotation for the orientation for mega tag 2 just like we do in disabled.  Why will it work there but not here?

Reply:
Ahh sorry I understand now, I think that makes sense then, but we probably want to add some hysteresis for the for the clear. clear the fault once we have updated our position with the MegaTag1 rotation? What do you think?

Reply:
That probably makes sense.  One thing I was debating was whether we should reset the pigeon as well.